cmake/onnxruntime_providers_cuda.cmake

```diff
@@ -33,19 +33,21 @@
   )
 
 
-  if (onnxruntime_CUDA_MINIMAL)
-    set(onnxruntime_providers_cuda_shared_srcs "")
-  else()
+  if (NOT onnxruntime_CUDA_MINIMAL)
     file(GLOB_RECURSE onnxruntime_providers_cuda_cu_srcs CONFIGURE_DEPENDS
       "${ONNXRUNTIME_ROOT}/core/providers/cuda/*.cu"
       "${ONNXRUNTIME_ROOT}/core/providers/cuda/*.cuh"
     )
+  else()
+    set(onnxruntime_providers_cuda_cu_srcs
+        "${ONNXRUNTIME_ROOT}/core/providers/cuda/math/unary_elementwise_ops_impl.cu"
```

Comment:
It seems we need to unary_elementwise_impl.cuh as well?
The definition of `UnaryElementWiseImpl()` is in that file and used by `cuda::Impl_Cast<SrcT, DstT>` which TRT EP will call to cast DOUBLE <- -> FLOAT or INT64 <-->INT32.

Reply:
The include should be added as the inlclude directory is set correctly. If you want I can add it to sources, but usually this is not the case in CMake I would say.

Reply:
yeah true, then no need to add it to sources